Makefile

```diff
@@ -0,0 +1,128 @@
+# ====================================================================================
+# Setup Project
+PROJECT_NAME := provider-http
+PROJECT_REPO := github.com/arielsepton/$(PROJECT_NAME)
+
+PLATFORMS ?= linux_amd64 linux_arm64
+
+# -include will silently skip missing files, which allows us
+# to load those files with a target in the Makefile. If only
+# "include" was used, the make command would fail and refuse
+# to run a target until the include commands succeeded.
+-include build/makelib/common.mk
+
+# ====================================================================================
+# Setup Output
+
+-include build/makelib/output.mk
+
+# ====================================================================================
+# Setup Go
+
+# Set a sane default so that the nprocs calculation below is less noisy on the initial
+# loading of this file
+NPROCS ?= 1
+
+# each of our test suites starts a kube-apiserver and running many test suites in
+# parallel can lead to high CPU utilization. by default we reduce the parallelism
+# to half the number of CPU cores.
+GO_TEST_PARALLEL := $(shell echo $$(( $(NPROCS) / 2 )))
+
+GO_STATIC_PACKAGES = $(GO_PROJECT)/cmd/provider
+GO_SUBDIRS += cmd internal apis
+GO111MODULE = on
+GOLANGCILINT_VERSION = 1.51.2
+-include build/makelib/golang.mk
+
+# ====================================================================================
+# Setup Kubernetes tools
+KIND_VERSION = v0.18.0
+UP_VERSION = v0.17.0
+UPTEST_VERSION = v0.5.0
+UP_CHANNEL = stable
+USE_HELM3 = true
+-include build/makelib/k8s_tools.mk
+
+# ====================================================================================
+# Setup Images
+
+IMAGES = provider-http
+-include build/makelib/imagelight.mk
+
+# ====================================================================================
+# Targets
+
+# run `make help` to see the targets and options
+
+# We want submodules to be set up the first time `make` is run.
+# We manage the build/ folder and its Makefiles as a submodule.
+# The first time `make` is run, the includes of build/*.mk files will
+# all fail, and this target will be run. The next time, the default as defined
+# by the includes will be run instead.
+fallthrough: submodules
+	@echo Initial setup complete. Running make again . . .
+	@make
+
+# ====================================================================================
+# Setup XPKG
+
+XPKG_REG_ORGS ?= xpkg.upbound.io/crossplane-contrib index.docker.io/crossplanecontrib

```

Comment:
Pushing to Docker Hub is more of a compatibility thing for older providers before `xpkg.upbound.io` to not break systems that may still be pulling from Docker's registry. It's not necessary here 👍🏼 

As a side note the repository should be all set and we'll set up the GitHub secrets to get this publishing!

Reply:
Awesome, thanks for the clarification! I’m looking forward to setting up the GitHub secrets for publishing. If there’s anything specific I need to do or be aware of during this setup, please feel free to let me know. 😊